src/utils/hashes/SHAMap/innerNode.ts

```diff
@@ -0,0 +1,123 @@
+import hashPrefix from '../hashPrefix'
+import sha512Half from '../sha512Half'
+
+import Leaf from './leafNode'
+import { NodeType, Node } from './node'
+
+const HEX_ZERO =
+  '0000000000000000000000000000000000000000000000000000000000000000'
+
+const SLOT_MAX = 15
+const HEX = 16
+
+/**
+ * Class for SHAMap InnerNode.
+ */
+class InnerNode extends Node {
+  public leaves: { [slot: number]: Node | undefined }
```

Comment:
Isn't e.g. `3: undefined` the same as `3` not being in the map at all? What's the importance in putting `undefined` there?

Reply:
That is correct, it is denoting that a leaf can either be `undefined` meaning nothing, or a `Node`, meaning there is a leaf there. 

This just formalizes this in typescript, so we can check if exists when inserting nodes...